### ISSUE TITLE
Bug: False positive on prefer-const (catch clause)

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno_lint#346]
```js
try {
  foo();
} catch (e) {
  console.error(e);
}
```

This should be okay but we're getting an error like

```
---- rules::prefer_const::tests::prefer_const_valid stdout ----
thread 'rules::prefer_const::tests::prefer_const_valid' panicked at 'Unexpected diagnostics found:
[
    LintDiagnostic {
        range: Range {
            start: Position {
                line: 1,
                col: 22,
            },
            end: Position {
                line: 1,
                col: 23,
            },
        },
        filename: "deno_lint_test.tsx",
        message: "\'e\' is never reassigned. Use \'const\' instead",
        code: "prefer-const",
        hint: None,
    },
]

source:
try { foo(); } catch (e) {}
```
